### ISSUE TITLE
Debug Wand causing a kick with a big network

### ISSUE THREAD

**Issue by inklit** [SquidDev-CC/CC-Tweaks#117]
The debug wand causes this error and kicks the player that attempted to use it.
Funniest part, you can't even rejoin. It keeps kicking you until the network is gone or the server restarts.

```
[17:52:09] [Server thread/INFO]: BTC has logged in.
[17:52:11] [Netty Epoll Server IO #0/ERROR] [FML]: NetworkDispatcher exception
java.lang.IllegalArgumentException: Payload may not be larger than 1048576 bytes
        at net.minecraft.network.play.server.S3FPacketCustomPayload.<init>(S3FPacketCustomPayload.jav
a:27) ~[gg.class:?]
        at net.minecraftforge.fml.common.network.internal.FMLProxyPacket.toS3FPackets(FMLProxyPacket.
java:136) ~[FMLProxyPacket.class:?]
        at net.minecraftforge.fml.common.network.handshake.NetworkDispatcher.write(NetworkDispatcher.
java:510) ~[NetworkDispatcher.class:?]
        at io.netty.channel.AbstractChannelHandlerContext.invokeWrite(AbstractChannelHandlerContext.j
ava:658) ~[AbstractChannelHandlerContext.class:?]
        at io.netty.channel.AbstractChannelHandlerContext.write(AbstractChannelHandlerContext.java:71
6) ~[AbstractChannelHandlerContext.class:?]
        at io.netty.channel.AbstractChannelHandlerContext.writeAndFlush(AbstractChannelHandlerContext
.java:706) ~[AbstractChannelHandlerContext.class:?]
        at io.netty.channel.AbstractChannelHandlerContext.writeAndFlush(AbstractChannelHandlerContext
.java:741) ~[AbstractChannelHandlerContext.class:?]
        at io.netty.channel.DefaultChannelPipeline.writeAndFlush(DefaultChannelPipeline.java:895) ~[D
efaultChannelPipeline.class:?]
        at io.netty.channel.AbstractChannel.writeAndFlush(AbstractChannel.java:240) ~[AbstractChannel
.class:?]
        at net.minecraft.network.NetworkManager$4.run(NetworkManager.java:249) [ek$4.class:?]
        at io.netty.util.concurrent.SingleThreadEventExecutor.runAllTasks(SingleThreadEventExecutor.j
ava:380) [SingleThreadEventExecutor.class:?]
        at io.netty.channel.epoll.EpollEventLoop.run(EpollEventLoop.java:268) [EpollEventLoop.class:?
]
        at io.netty.util.concurrent.SingleThreadEventExecutor$2.run(SingleThreadEventExecutor.java:11
6) [SingleThreadEventExecutor$2.class:?]
        at java.lang.Thread.run(Thread.java:745) [?:1.8.0_91]
[17:52:11] [Server thread/INFO]: BTC lost connection: Internal Exception: java.lang.IllegalArgumentEx
ception: Payload may not be larger than 1048576 bytes
[17:52:11] [Server thread/INFO]: BTC has disconnected.
```

**Comment by SquidDev** [comment]
-  Compress nodes
- Only include names of nodes within set number of blocks and update them.
- Radius limit on sending packets
- Remove watcher on player disconnect

**Comment by SquidDev** [comment]
Fixed in 27515877ce324e412a6567d773b964729eadbf16

Amusing image of 40k nodes:

![](http://i.imgur.com/3uqz5q2.png?v=2)
